### PR TITLE
Change release to trigger on version change

### PR DESCRIPTION
**Pull request by idastambuk**
Changing the release step to trigger on every commit to main with a version change. 

Basically copied the steps from [grafana/aws-sdk-react,](https://github.com/grafana/grafana-aws-sdk-react/blob/main/.github/workflows/publish-npm.yml) and replaced the github release step from the old publish as a separate job. Lmk if it looks good

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: main
+      # limit releases to version changes - https://github.com/EndBug/version-check
+      - name: Check version changes
+        uses: EndBug/version-check@v1
+        id: version_check
+        with:
+          file-url: https://unpkg.com/@grafana/aws-sdk@latest/package.json
```
Comment (kevinwcyu) [inline .github/workflows/release.yml:21]: URL should point to `@grafana/async-query-data` package

**Review by kevinwcyu — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        with:
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@grafana'
+
+      - name: Publish package to Github Packages
+        if: steps.version_check.outputs.changed == 'true'
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (kevinwcyu) [inline .github/workflows/release.yml:61]: i don't actually see a `GITHUB_TOKEN` in this repo's settings. do we need to add that

File: .github/workflows/release.yml (hunk 1 above)
Comment (idastambuk) [reply .github/workflows/release.yml:21]: True lol 🤦🏻‍♀️

File: .github/workflows/release.yml (hunk 2 above)
Comment (idastambuk) [reply .github/workflows/release.yml:61]: The GITHUB_TOKEN[ is a temporary secret](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) that is created at runtime and has certain limited permissions, so we don't need to explicitly add it. I think this should be ok without giving it any specific permissions for this action🤞🏻

**Pull request merged**